express-app/src/index.js

```diff
@@ -0,0 +1,176 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
+const cors = require('cors');
+const crypto = require('crypto');
+const { validationResult } = require('express-validator');
+const cookie = require('cookie');
+require('dotenv').config();
+const config = require('./config');
+const jwt = require('jsonwebtoken')
+
+mongoose.connect('mongodb://mongo:27017', {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+})
+  .then(() => console.log('Connected to faizs database'))
+  .catch(err => console.log(err));
+
+const app = express();
+app.use(cors({ origin: 'http://localhost:3000', credentials: true }));
+app.use(bodyParser.json());
+
+const User = mongoose.model('User', new mongoose.Schema({
+  username: {
+    type: String,
+    required: true,
+    unique: true,
+  },
+  encryptedPassword: {
+    type: String,
+    required: true,
+  },
+  salt: {
+    type: String,
+    required: true,
+  },
+}));
+
+function hashPassword(password, salt) {
+  const key = crypto.pbkdf2Sync(password, salt + config.secretKey, 1000, 64, 'sha512');
```

Comment:
Can you explain the value of adding a secret to the salt?

Reply:
It only allows the access to the salt when you know the secret which improves the security.